src/main/java/org/jitsi/videobridge/OctoEndpoints.java

```diff
@@ -0,0 +1,291 @@
+/*
+ * Copyright @ 2018 Atlassian Pty Ltd
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.jitsi.videobridge;
+
+import org.jitsi.impl.neomedia.rtp.*;
+import org.jitsi.service.neomedia.*;
+import org.jitsi.util.*;
+
+import java.io.*;
+import java.util.*;
+import java.util.stream.*;
+
+/**
+ * Manages the list of remote/foreign/Octo endpoints for a specific
+ * {@link Conference}.
+ *
+ * @author Boris Grozev
+ */
+public class OctoEndpoints
+{
+    /**
+     * The {@link Logger} used by the {@link RtpChannel} class to print debug
+     * information. Note that instances should use {@link #logger} instead.
+     */
+    private static final Logger classLogger
+        = Logger.getLogger(OctoChannel.class);
+
+    /**
+     * The owning conference.
+     */
+    private Conference conference;
+
+    /**
+     * The conference's Octo channel for audio, if there is one, or {@code null}.
+     */
+    private OctoChannel audioChannel;
+
+    /**
+     * The conference's Octo channel for video, if there is one, or {@code null}.
+     */
+    private OctoChannel videoChannel;
+
+    /**
+     * Used to synchronize access in this class. Note that some of the
+     * operations access the {@link Conference}'s list of endpoints which
+     * uses a separate lock. Therefore, to avoid possible deadlocks this
+     * class' API MUST NOT be called while holding a lock on the conference
+     * endpoints list.
+     */
+    private final Object syncRoot = new Object();
```

Comment:
if this is on the class level, couldn't `this` be used?

Reply:
if not, can we rename it to something that makes it clear the intended scope of what this lock covers? i.e. if i'm adding new code, how can we make it clear whether or not i should acquire this lock when modifying data.

Reply:
also, if we use `this`, then some of the methods (which are entirely synchronized) can change from:

```
private final Foo method(...) {
    synchronized(syncRoot) {
        // code
    }
}
```
to
```
synchronized private final Foo method(...) {
    // code
}
```

Reply:
As a rule of thumb I avoid synchronizing on `this` and use a private object (to make it impossible for someone use the same lock in another class). I'll clarify the doc.